posthog/tasks/test/test_update_cache.py

```diff
@@ -0,0 +1,77 @@
+from posthog.tasks.update_cache import update_cache_item, update_cached_items
+from posthog.api.test.base import BaseTest
+from posthog.models import Filter, DashboardItem, Dashboard, Funnel
+from posthog.utils import generate_cache_key
+from django.core.cache import cache
+from freezegun import freeze_time
+from unittest.mock import patch, MagicMock
+from django.utils.timezone import now
+import json
+
+
+class TestUpdateCache(BaseTest):
+    TESTS_API = True
+
+    @patch("posthog.tasks.update_cache.group.apply_async")
+    @patch("posthog.tasks.update_cache.update_cache_item.s")
+    def test_refresh_dashboard_cache(
+        self, patch_update_cache_item: MagicMock, patch_apply_async: MagicMock
+    ) -> None:
+        # There's two things we want to refresh
+        # Any shared dashboard, as we only use cached items to show those
+        # Any dashboard accessed in the last 7 days
+        filter_dict = {
+            "events": [{"id": "$pageview"}],
+            "properties": [{"key": "$browser", "value": "Mac OS X"}],
+        }
+        filter = Filter(data=filter_dict)
+        shared_dashboard = Dashboard.objects.create(team=self.team, is_shared=True)
+        funnel = Funnel.objects.create(
+            team=self.team,
+            name="funnel",
+            filters={
+                "events": [{"id": "user signed up", "type": "events", "order": 0},],
+            },
+        )
+        item = DashboardItem.objects.create(
+            dashboard=shared_dashboard, filters=filter.to_dict(), team=self.team
+        )
+        funnel_item = DashboardItem.objects.create(
+            dashboard=shared_dashboard, funnel=funnel, team=self.team
+        )
+
+        dashboard_to_cache = Dashboard.objects.create(
+            team=self.team, is_shared=True, last_accessed_at=now()
+        )
+        item_to_cache = DashboardItem.objects.create(
+            dashboard=dashboard_to_cache,
+            filters={"events": [{"id": "cache this"}]},
+            team=self.team,
+        )
+
+        dashboard_do_not_cache = Dashboard.objects.create(
+            team=self.team, is_shared=True, last_accessed_at="2020-01-01"
+        )
+        item_do_not_cache = DashboardItem.objects.create(
+            dashboard=dashboard_to_cache,
```

Comment:
Should this be added to `dashboard_do_not_cache` ? Maybe not, I'm not sure of the intent here. In any case, `dashboard_do_not_cache` is not used after it's defined.

Reply:
Yep